### DISCUSSION TITLE
How does prisma cloud handle user’s connection strings?

### DISCUSSION THREAD

**Discussion by zomars** [prisma/prisma#17847] [Q&A]
At @calcom we're implementing a way for users to save their encrypted connection string with us and allow them to manage it from our public API.

Similarly on how you do it in Prisma Cloud. 

The thing is we are hitting connection pool limits and having mixed data for certain requests (not good). Tried disconnecting and creating new instances but it doesn't work. 

How do you handle this in Prisma Cloud?

**Comment by nurul3101** [comment]
Hello @zomars 👋 

To confirm, are you getting connection timeouts when connecting to the user's databases?
Can you elaborate on what you mean by getting mixed data for certain requests?

I am not aware of the exact implementation, but I can ask our Prisma Cloud engineering team if they have any insights for this problem.

**Reply by zomars** [reply]
We use our DB connection string by default in our API. But for enterprise customers we create a custom instance with their own DB url. There were some cases where the default prisma was still being used and customers we're seeing OUR data instead of theirs.

**Reply by zomars** [reply]
The code is closed source but here's the helper we use to setup a custom prisma instance so far:

https://raw.githubusercontent.com/calcom/api/main/lib/helpers/customPrisma.ts?token=GHSAT0AAAAAABRLGCK3KT2P64XTDKHII6SYY7G7SQQ

**Reply by janpio** [reply]
That link returns a 404 for me.

What you describe should not be possible, unless Prisma is somehow configured to know both connection string at different times.

Best bring this up in our shared, private Slack channel - then we can also mention private code snippets etc.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
